### PR TITLE
SSL security for HDP cluster

### PR DESCRIPTION
**Pull request by bhushankandalkar88**

### CODE REVIEW

**Comment by razorsedge** [comment]
@bhushankandalkar88 I see Hortonworks talk about `/etc/security/keytabs` in https://docs.hortonworks.com/HDPDocuments/HDP2/HDP-2.6.5/bk_security/bk_security.pdf and other docs.  I also see references to `/etc/security/serverKeys` and `/etc/security/clientKeys` in https://docs.hortonworks.com/HDPDocuments/HDP2/HDP-2.6.5/bk_security/content/ch_wire-webhdfs-mr-yarn.html .  I'm just not happy with `/opt/hortonworks/security`.

Even [this](https://docs.hortonworks.com/HDPDocuments/HDP3/HDP-3.1.0/configuring-wire-encryption/content/installing_certificates_in_the_hadoop_ssl_keystore_factory_hdfs_mapreduce_and_yarn.html)  is fairly vague.

**Comment by bhushankandalkar88** [comment]
@razorsedge Thanks for reviewing. Instead of '/opt/hortonworks/security', I will use '/etc/security/keytabs' as an directory structure.

**Comment by razorsedge** [comment]
Part of the problem there is that `/etc/security` is already used by the operating system.  I can see putting keytabs in `/etc/security/keytabs` as that would align with Hortonworks' documentation, but placing x509 certs in there as well just feels weird.

Lets just do `/etc/hortonworks/security`.

File: tls/create_security_hdp.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright Clairvoyant 2019
+
+echo "********************************************************************************"
+echo "*** $(basename "$0")"
+echo "********************************************************************************"
+echo "Making TLS security directories..."
+mkdir -p -m 0755 /opt/hortonworks/security
```
Comment (razorsedge) [inline tls/create_security_hdp.sh]: Lets just do `/etc/hortonworks/security` .

File: tls/generate_csr_hdp.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  . /etc/profile.d/java.sh
+fi
+
+if [ -f /opt/hortonworks/security/jks/localhost-keystore.jks ]; then
+  echo "ERROR: Keystore already exists.  Exiting..."
+  exit 1
+fi
+
+keytool -genkeypair -alias localhost -keyalg RSA -sigalg SHA256withRSA \
+ -keystore /opt/hortonworks/security/jks/localhost-keystore.jks \
```
Comment (razorsedge) [inline tls/generate_csr_hdp.sh]: Lets just do `/etc/hortonworks/security` .

File: tls/import_cert_hdp.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+echo "Importing signed TLS certificate and chain..."
+if [ -f /etc/profile.d/jdk.sh ]; then
+  . /etc/profile.d/jdk.sh
+elif [ -f /etc/profile.d/java.sh ]; then
+  . /etc/profile.d/java.sh
+fi
+
+keytool -importcert -trustcacerts -noprompt -alias RootCA \
+-keystore /opt/hortonworks/security/jks/localhost-keystore.jks \
```
Comment (razorsedge) [inline tls/import_cert_hdp.sh]: Lets just do `/etc/hortonworks/security` .

File: tls/install_rootCA_hdp.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  JAVA_HOME=/usr/java/default
+fi
+
+if [ -z "${JAVA_HOME}" ]; then echo "ERROR: \$JAVA_HOME is not set."; exit 10; fi
+
+if [ ! -f "${JAVA_HOME}"/jre/lib/security/jssecacerts ]; then
+  #TODO: On el7: /usr/java/default/jre/lib/security/cacerts -> /etc/pki/java/cacerts
+  /bin/cp -p "${JAVA_HOME}"/jre/lib/security/cacerts "${JAVA_HOME}"/jre/lib/security/jssecacerts
+fi
+keytool -importcert -file /opt/hortonworks/security/CAcerts/ca.cert.pem -alias CAcert -keystore "${JAVA_HOME}"/jre/lib/security/jssecacerts -storepass changeit -noprompt -trustcacerts
```
Comment (razorsedge) [inline tls/install_rootCA_hdp.sh]: Lets just do `/etc/hortonworks/security` .

File: tls/install_rootCA_hdp.sh [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright Clairvoyant 2015
+
+# Function to discover basic OS details.
+discover_os() {
```
Comment (razorsedge) [inline tls/install_rootCA_hdp.sh]: `discover_os()` has been updated in `master`.  Please apply those changes here.

**Comment by bhushankandalkar88** [comment]
@razorsedge made and committed suggested changes. Please review it: https://github.com/teamclairvoyant/hadoop-deployment-bash/pull/58/commits/6119862fafd3e101a5da2946c849f1e039567aaf

**Pull request merged**